src/cli/rustup_mode.rs

```diff
@@ -987,17 +1007,19 @@ fn update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
 
             let status = if !toolchain.is_custom() {
                 let components: Vec<_> = m
-                    .values_of("components")
-                    .map(|v| v.collect())
-                    .unwrap_or_else(Vec::new);
+                    .try_get_many::<String>("components")
+                    .ok()
+                    .flatten()
+                    .map_or_else(Vec::new, |v| v.map(|s| &**s).collect());
                 let targets: Vec<_> = m
-                    .values_of("targets")
-                    .map(|v| v.collect())
-                    .unwrap_or_else(Vec::new);
+                    .try_get_many::<String>("targets")
+                    .ok()
+                    .flatten()
+                    .map_or_else(Vec::new, |v| v.map(|s| &**s).collect());
                 let distributable = DistributableToolchain::new(&toolchain)?;
                 Some(distributable.install_from_dist(
-                    m.is_present("force"),
-                    m.is_present("allow-downgrade"),
+                    m.get_flag("force"),
+                    matches!(m.try_get_one::<bool>("allow-downgrade"), Ok(Some(true))),
```

Comment:
I think this reads better than `m.try_get_one::<bool>("allow-downgrade").ok().flatten().copied().unwrap_or_default()`